sdk/personalizer/Azure.AI.Personalizer/src/Models/PersonalizerClientProperties.Serialization.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.AI.Personalizer.Models
+{
+    internal partial class PersonalizerClientProperties
+    {
+        internal static PersonalizerClientProperties DeserializePersonalizerServiceProperties(JsonElement element)
```

Comment:
Can you explain why we need a custom deserializer instead of using `JsonSerializer.Deserialize<PersonalizerClientProperties>(string)`?